Microsoft.Toolkit.Uwp.UI.Controls/Carousel/CarouselItemAutomationPeer.cs

```diff
@@ -0,0 +1,147 @@
+﻿// Licensed to the .NET Foundation under one or more agreements.
+// The .NET Foundation licenses this file to you under the MIT license.
+// See the LICENSE file in the project root for more information.
+
+using Microsoft.Toolkit.Uwp.UI.Controls;
+using Windows.UI.Xaml.Automation.Peers;
+using Windows.UI.Xaml.Automation.Provider;
+
+namespace Microsoft.Toolkit.Uwp.UI.Automation.Peers
+{
+    /// <summary>
+    /// Defines a framework element automation peer for the <see cref="CarouselItem"/>.
+    /// </summary>
+    public class CarouselItemAutomationPeer : FrameworkElementAutomationPeer, ISelectionItemProvider
+    {
+        /// <summary>
+        /// Initializes a new instance of the <see cref="CarouselItemAutomationPeer"/> class.
+        /// </summary>
+        /// <param name="owner">
+        /// The <see cref="CarouselItem" /> that is associated with this <see cref="T:Windows.UI.Xaml.Automation.Peers.CarouselItemAutomationPeer" />.
+        /// </param>
+        public CarouselItemAutomationPeer(CarouselItem owner)
+            : base(owner)
+        {
+        }
+
+        /// <summary>Gets a value indicating whether an item is selected.</summary>
+        /// <returns>True if the element is selected; otherwise, false.</returns>
+        public bool IsSelected => this.OwnerCarouselItem.IsSelected;
+
+        /// <summary>Gets the UI Automation provider that implements ISelectionProvider and acts as the container for the calling object.</summary>
+        /// <returns>The UI Automation provider.</returns>
+        public IRawElementProviderSimple SelectionContainer
+        {
+            get
+            {
+                Carousel parent = this.OwnerCarouselItem.ParentCarousel;
+                if (parent == null)
+                {
+                    return null;
+                }
+
+                AutomationPeer peer = FromElement(parent);
+                return peer != null ? this.ProviderFromPeer(peer) : null;
+            }
+        }
+
+        private CarouselItem OwnerCarouselItem
+        {
+            get { return this.Owner as CarouselItem; }
+        }
+
+        /// <summary>Adds the current element to the collection of selected items.</summary>
+        public void AddToSelection()
+        {
+            CarouselItem owner = this.OwnerCarouselItem;
+            Carousel parent = owner.ParentCarousel;
+            parent?.SetSelectedItem(owner);
+        }
+
+        /// <summary>Removes the current element from the collection of selected items.</summary>
+        public void RemoveFromSelection()
```

Comment:
So UIA users can reach a state that isn't reachable through mouse interaction? Or is there a way to unselect an item as user?

Reply:
Good point. The `Carousel` doesn't provide the user functionality for unselecting an item. Easily sorted.

Reply:
This is sorted now, removed the code for this.